langchain/text_splitter.py

```diff
@@ -439,3 +440,261 @@ def __init__(self, **kwargs: Any):
             "",
         ]
         super().__init__(separators=separators, **kwargs)
+
+
+class Language(str, Enum):
+    CPP = "cpp"
+    GO = "go"
+    JAVA = "java"
+    JS = "js"
+    PHP = "php"
+    PROTO = "proto"
+    PYTHON = "python"
+    RST = "rst"
+    RUBY = "ruby"
+    RUST = "rust"
+    SCALA = "scala"
+    SWIFT = "swift"
+
+
+class CodeTextSplitter(RecursiveCharacterTextSplitter):
+    def __init__(self, language: Language, **kwargs: Any):
+        """
+        A generic code text splitter supporting many programming languages.
```

Comment:
why not include markdown and latex?  They are already here, could still expose those classes for backwards compatibility but would also make sense to consolidate them here.

Reply:
Sounds great! I have added them